Packs/OneLogin/Integrations/OneLoginEventCollector/OneLoginEventCollector.py

```diff
@@ -0,0 +1,367 @@
+        Send request to get event types from OneLogin.
+        """
+        raw_response = self._http_request(method='GET', url_suffix='/api/1/events/types')
+        check_response(raw_response)
+
+        demisto.debug('Succesfully got event types.')
+        return raw_response.get('data', [])
+
+    def get_event_types_from_last_run(self, last_run: dict):
+        """
+        Gets the event types from the LastRun obj.
+        For the first fetch, will send a call to get the event types from OneLogin and will update the LastRun as well.
+        """
+
+        if 'event_types' not in last_run:
+            event_types = self.get_event_types_request()
+            last_run['event_types'] = {str(event_type['id']): event_type['name'] for event_type in event_types}
+
+        return last_run.get('event_types')
+
+    def convert_type_id_to_name(self, event: dict, event_types: dict, last_run: dict) -> str:
+        """
+        Gets the event type name by the event type id.
+
+        Args:
+            event: The event getting from the OneLogin events API call.
+            event_types: Event types getting from the LastRun obj.
+            last_run: The LastRun obj.
+
+        Returns:
+            (str): The event type name.
+        """
+
+        event_type_name = ''
+        if event_type_name := get_event_type_name(str(event['event_type_id']), event_types):
+            return event_type_name
+
+        demisto.debug(f"Could not find the event type id {str(event['event_type_id'])}. "
+                      f"Trying to request the event types from OneLogin.")
+        event_types_res = self.get_event_types_request()
+        event_types = {str(event_type['id']): event_type['name'] for event_type in event_types_res}
+
+        if event_type_name := get_event_type_name(str(event['event_type_id']), event_types):
+            last_run['event_types'] = event_types
+            return event_type_name
+
+        demisto.debug(f"Could not find the event type id '{str(event['event_type_id'])}' "
+                      f"in the event types from OneLogin, returning an empty value")
+        return ''
+
+    def get_access_token(self):
+        """
+        Gets the API access token.
+        """
+
+        int_context = get_integration_context()
+        if 'access_token' in int_context:
+            expired_token_date_time = parse(int_context['expired_token_time'], settings={'TIMEZONE': 'UTC'})
+            if expired_token_date_time and datetime.now(timezone.utc) - expired_token_date_time < timedelta(minutes=590):
+                demisto.debug("Found access token in integration context.")
+                return int_context.get('access_token')
+
+        demisto.debug("Access token has expired, sending request to get a new access token.")
+        access_token = self.get_access_token_request()
+        set_integration_context({
+            'access_token': access_token,
+            'expired_token_time': datetime.utcnow().strftime(DATE_FORMAT)
```

Comment:
this line makes it look like expiry time is now (confusing), I'd do:
```suggestion
            'expired_token_time': datetime.utcnow().strftime(DATE_FORMAT) + timedelta(minutes=590)
```
and remove the timedelta above

Reply:
done